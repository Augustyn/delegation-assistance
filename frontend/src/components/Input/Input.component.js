import React from "react";
import { array, func, string, oneOfType } from "prop-types";
import { Field } from "redux-form";

import RenderInput from "../renderers/RenderInput/RenderInput.renderer";
import RenderTextarea from "../renderers/RenderTextarea/RenderTextarea.renderer";
import RenderSelect from "../renderers/RenderSelect/RenderSelect.renderer";
import RenderDateTimePicker from "../renderers/RenderDateTimePicker/RenderDateTimePicker.renderer";
import RenderTypeahead from "../renderers/RenderTypeahead/RenderTypeahead.renderer";
import RenderFilePond from "../renderers/RenderFilePond/RenderFilePond.renderer.js";

import styles from "./Inputs.module.scss";

const components = {
  input: RenderInput,
  textarea: RenderTextarea,
  select: RenderSelect,
  datepicker: RenderDateTimePicker,
  filepond: RenderFilePond,
  typeahead: RenderTypeahead
};

const Input = ({ name, component, label, placeholder, validate, classes, ...rest }) => {
  return (
    <div className={[styles.field, classes].join(" ")}>
      <div className={styles.label}>
        {label ? (
          <label className={styles["label-bold"]} htmlFor={"input_" + name}>
            {label}
          </label>
        ) : null}
        <Field
          {...rest}
          id={"input_" + name}
          className={"input"}
          name={name}
          validate={validate}
          label={placeholder}
          component={components[component] || RenderInput}
        />
      </div>
    </div>
  );
};

Input.defaultProps = {
  component: "input"
};

Input.propTypes = {
  classes: string,
  component: string.isRequired,
  label: string,
  name: string.isRequired,
  options: array,
  placeholder: string,
  type: string,
  validate: oneOfType([func, array])
};

export default Input;
