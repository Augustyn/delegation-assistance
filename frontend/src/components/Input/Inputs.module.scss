@import "../../static/styles/constants";

.render-field {
  .input[type="text"],
  .input[type="password"] {
    display: block;
    width: 100%;
    height: calc(2em + 0.75rem + 2px);
    padding: 0.5rem 0.75rem;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
    outline: none;
    margin: 5px 0px;

    &.invalid {
      color: $error-color;
      border-color: rgba($error-color, 0.5);
      box-shadow: 0 0 0 0.1rem rgba($error-color, 0.25);
    }

    &:focus {
      border-color: rgba($primary-color, 0.5);
      box-shadow: 0 0 0 0.1rem rgba($primary-color, 0.25);
    }
  }
}

.field{
  width: 100%;
}

.label {
  &.label-fluid {
    width: 100%;
  }
}

.label-bold {
  font-weight: bold;
  color: $primary-color;
}

form {
  margin: 10px;
}
