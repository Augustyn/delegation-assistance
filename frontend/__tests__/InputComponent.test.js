import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Input from "../src/components/Input/Input.component";

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    name: "testName",
    label: "testLabel",
  };

  const enzymeWrapper = shallow(<Input {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe("Input Component", () => {
  it("should render Input correctly", () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it("should render provided label correctly", () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find(".label-bold").props().children).toEqual(
      "testLabel"
    );
  });

  it("should change disabled state", ()=>{
    const { props } = setup();
    props.disabled = false;
    const enzymeWrapper = shallow(<Input {...props} />);
    expect(enzymeWrapper.find("Field").prop("disabled")).toEqual(false);
    props.disabled = true;
    enzymeWrapper.setProps(props);
    expect(enzymeWrapper.find("Field").prop("disabled")).toEqual(true);
  });

});
