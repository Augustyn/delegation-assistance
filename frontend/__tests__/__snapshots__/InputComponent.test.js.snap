// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render Input correctly 1`] = `
<div
  className=" "
>
  <div>
    <label
      className=" "
    >
      <span
        data-test="label-value"
      >
        testLabel
      </span>
      <Field
        className="input"
        component={[Function]}
        disabled={false}
        label=""
        minlength={0}
        name="testName"
        type="text"
        validate={null}
        value=""
      />
    </label>
  </div>
</div>
`;
